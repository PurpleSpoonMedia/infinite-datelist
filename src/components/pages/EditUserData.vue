<template>
  <div>
    <h3 class="text-center">{{ user.name }}</h3>
    <app-user-form
      :userProp="user"
      formType="editUser"
      @userData="updateUser"
    ></app-user-form>
  </div>
</template>

<script>
import UserForm from '../UserForm';
export default {
  methods: {
    updateUser(editedUser) {
      this.$store.dispatch('editUser', editedUser);
    }
  },
  computed: {
    user() {
      const user = this.$store.state.users.find(user => user.id === this.$route.params.id);
      return {
        ...user,
        services: user.services.split(', '),
      }
    }
  },
  components: {
    appUserForm: UserForm
  }
}
</script>

<style lang="sass" scoped>

</style>