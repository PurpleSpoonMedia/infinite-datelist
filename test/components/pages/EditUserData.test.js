import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import EditUserData from '../../../src/components/pages/EditUserData.vue';
import users from '../../fixtures/users';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('EditUserData.vue', () => {
  let store, state;

  beforeEach(() => {
    state = {
      users,
    };
    store = new Vuex.Store({
      state
    });
  })

  test('should render EditUserData correctly', () => {
    const wrapper = shallowMount(EditUserData, {
      store,
      localVue,
      mocks: {
        $route: {
          params: { id: 1 }
        }
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should call updateUser method when userData is emitted', () => {
    const updateUser = jest.fn();
    const wrapper = shallowMount(EditUserData, {
      store,
      localVue,
      mocks: {
        $route: {
          params: { id: 1 }
        }
      }
    });

    wrapper.vm.$on('userData',  updateUser);
    wrapper.vm.$emit('userData', users[0]);

    expect(updateUser).toHaveBeenLastCalledWith(users[0]);
  })
});
