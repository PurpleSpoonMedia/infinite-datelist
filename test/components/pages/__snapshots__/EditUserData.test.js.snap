// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditUserData.vue should render EditUserData correctly 1`] = `
<div>
  <h3 class="text-center">Test 1</h3>
  <appuserform-stub userprop="[object Object]" formtype="editUser"></appuserform-stub>
</div>
`;
